### PR TITLE
merlin-sdk docker image

### PR DESCRIPTION
**Pull request by ariefrahmansyah**
**What this PR does / why we need it**:

This PR introduces new Docker image, namely `merlin-sdk` that contains the SDK source code and its test suite. This docker enables faster development iteration.

This PR also updates mock_oauth to cater for running e2e-test on machine that use Google Workload Identity.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Checklist**

- [ ] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [ ] Updated documentation
- [ ] Update Swagger spec if the PR introduce API changes
- [ ] Regenerated Golang and Python client if the PR introduce API changes

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -41,6 +41,25 @@ jobs:
...
+      matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
+    needs:
+      - publish-python-sdk
+    steps:
+      - uses: actions/checkout@v2
+      - name: Build Merlin SDK Docker py${{ matrix.python-version }}
+        working-directory: ./python/sdk
+        run: |
+          IMAGE_TAG="ghcr.io/caraml-dev/merlin-sdk:py${{ matrix.python-version }}-${{ inputs.version }}"
```
Comment (krithika369) [inline .github/workflows/release.yml:57]: Shall we define the `CONTAINER_REGISTRY: ${{ env.DOCKER_REGISTRY }}/${{ github.repository }}` variable common to all jobs and create this `IMAGE_TAG` variable based on that? We are already doing this for some jobs like `publish-pyfunc`.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -41,6 +41,25 @@ jobs:
...
+          SDK_VERSION="$(sed 's/^v\(.*\)-/\1/' <<< "${{ inputs.version }}")"
+          docker login ${{ env.DOCKER_REGISTRY }} -u ${{ github.actor }} -p ${{ secrets.ghcr_token }}
```
Comment (krithika369) [inline .github/workflows/release.yml:59]: Why not use the `docker/login-action@v1` action for this like some of the other publish jobs?

File: python/sdk/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+LABEL org.opencontainers.image.source https://github.com/caraml-dev/merlin
+
+WORKDIR /root
+ENV PYTHONPATH /root
+
+RUN apt-get update && apt-get install build-essential curl vim wget -y
+
+COPY . .${WORKDIR}
+
+ARG VERSION==0.28.0
```
Comment (krithika369) [inline python/sdk/Dockerfile:13]: Shall we not set the default here to force setting the value during build?

File: python/sdk/test/client_test.py [hunk 4]
```diff
@@ -353,6 +353,22 @@ def mock_oauth():
         "aliases": ["default"]
     }
     """, status=200, content_type='application/json')
+
+    responses.add('GET',
```
Comment (krithika369) [inline python/sdk/test/client_test.py:357]: Thanks for this change! Btw, the `mock_oauth` fixture is defined in 2 files (this one and `merlin_test.py`). Shall we make it a common fixture in a new `fixtures.py` or something?

**Review by krithika369 — APPROVED** [review]
LGTM. Thanks for this, @ariefrahmansyah ! 🚀 

Left some small comments.

File: python/sdk/test/client_test.py (hunk 4 above)
Comment (ariefrahmansyah) [reply python/sdk/test/client_test.py]: Refactored by merging mock_oauth in client_test.py and merlin_test.py to one in conftest.py. Thanks for the suggestion!

**Pull request merged**